### ISSUE TITLE
gcc segfault with newly developed variant reflection

### ISSUE THREAD

**Issue by jbbjarnason** [stephenberry/glaze#604]
To begin with great efforts in adopting reflection on all compilers!

I was updating to `1.9.2` this morning and apparently `gcc` is crashing :smile: with the newly adopted reflection of variant it seems. Obviously this is a `gcc` bug but it would be nice if we could make workarounds.

Here is the call stack shown, https://github.com/Skaginn3x/framework/actions/runs/7139516819/job/19443006771?pr=335#step:6:621

I will try to make a minimal example to reproduce this issue.

For reference, there is no issue compiling the same code with clang.

**Comment by jbbjarnason** [comment]
So this is a problem with gcc 13, I added a PR with gcc 13 in github workflows here, https://github.com/stephenberry/glaze/pull/605 which reproduces the error.

@stephenberry do you have an account at gcc bugzilla or should I make one to report this?

**Comment by stephenberry** [comment]
@jbbjarnason, I discovered this last night as well, as my company uses gcc 13 in production. I do not currently have a bugzilla account, so feel free to go ahead and report this. I'm going to try to simplify the code this morning so that gcc will build. I was able to simplify the code last night so that the compiler no longer segfaults, but its errors are bogus.

**Comment by stephenberry** [comment]
Here is a simplified test that causes the segfault:
```c++
struct put_action
{
};

struct delete_action
{
};

using tagged_variant = std::variant<put_action, delete_action>;

template <>
struct glz::meta<tagged_variant>
{
   static constexpr std::string_view tag = "action";
   static constexpr auto ids = std::array{"PUT", "DELETE"}; // Defaults to glz::name_v of the type
};
```

**Comment by stephenberry** [comment]
I found a workaround, for some reason GCC 13 is corrupting the capture by reference for `std::array<std::string_view, get_max_keys<T, N>> data{};`

This is for the function: `get_combined_keys_from_variant`

The fix is that we can just change this to a pointer and GCC will properly capture the pointer:
```c++
auto* data_ptr = &data; // This intermediate pointer is necessary for GCC 13 (otherwise segfaults with reflection logic)
```

**Comment by stephenberry** [comment]
Fixed in #605

**Comment by jbbjarnason** [comment]
Great job
